src/utils/InputValidators.js

```diff
@@ -63,6 +79,21 @@ export function MacAddressValidator(macAddress) {
     retValue.errorMsg = translate('input.validator.macaddress.error');
   }
 
+  if(props && props.exist_mac_addresses && props.exist_mac_addresses.includes(macAddress)) {
+    retValue.isValid = false;
+    retValue.errorMsg = translate('input.validator.macaddress.exist.error');
+  }
+
+  if(props && props.exist_availservers_mac_addr_objs) {
+    let find =
+      props.exist_availservers_mac_addr_objs.find(mac => mac['mac-addr'] === macAddress);
+    if (find) {
```

Comment:
This logic is broken. `find()` returns the array element, so if the first element (`[0]`) were a match, this logic would not detect it. You should probably replace `find()` with `some()`, which returns `true` if there is a match (and also rename the `find` variable to something like `found`)

Reply:
my element is an array of objects.  I can not use some() since I want the object back so I can use the serverId to construct my message. I think the better way is to change if(find) to be if (found !== undefined)

Reply:
You are right, it was my misunderstanding. The code here is fine.